src/CMakeLists.txt

```diff
@@ -0,0 +1,42 @@
+set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
+set(THREADS_PREFER_PTHREAD_FLAG TRUE)
+find_package(Threads REQUIRED)
```

Comment:
We probably need to validate `CMAKE_USE_PTHREADS_INIT` too, as we're explicitly using pthread inside.

Reply:
Also, what do you think about separate cmake file containing library dependencies that would be included here and in config template?

Reply:
1. I wouldn't say we're explicitly using pthread inside, that flag will be used nonetheless. `THREADS_PREFER_PTHREAD_FLAG` is saying to look for this flag first, then look for other libraries. It changes search priority for flag from lowest to highest.
2. `CMAKE_THREAD_PREFER_PTHREAD` can be removed, I looked closely in module's code, and it's used only for IRIX systems. This var was also completely removed in future CMake versions (it was gone in 3.16).
3. I was thinking about making `FindMySQL` module, but I haven't done that, maybe in a future. For now, I don't think it's a great idea to create include file just for 2 lines of code, that must be exported as well.